analysis/table_2_new.R

```diff
@@ -47,9 +47,18 @@ table_2_calculation <- function(survival_data, event,cohort,subgrp, subgrp_level
   }
   
   # filter the population according to the subgrp level
-  data_active$active_subgrp_full_name <- data_active[,subgrp_full_name]
+  setnames(data_active,
+           old = paste0(subgrp_full_name),
+           new = "active_subgrp_full_name")
```

Comment:
Why to you change old to new (lines 50-52) and then old to new again (lines 58-60) with old being the former new and new being the former old. In that case, would it be easier to remove both changes and ensure that the set of names are the original setnames i.e. `subgrp_full_name` or `paste0(subgrp_full_name)` in the `data_active` dataset ?